### ISSUE TITLE
`JdbcBufferedConsumerFactory` to support checkpointing

### ISSUE THREAD

**Issue by ryankfu** [airbytehq/airbyte#23678]
## What
Currently `Jdbc` Destination connectors do not support checkpointing and if you publish the connector it will write data to `tmp` and mark as committed. To prevent this issue, we will need to update the consumer factory to checkpointing

## How
After this change all JDBC destinations (Postgres, MySQL, MS SQL, Oracle) will write directly to `_raw` tables and will not create `_airbyte_tmp` tables

**Comment by grishick** [comment]
Mid-sprint status: the change is blocked by a broken integration test. Ryan is working on fixing the test.

**Comment by ryankfu** [comment]
In-Progress:
- Get Demo to Destinations Team
- Write a dependency blocker for MySQL (cc: @ryankfu)

Future Work:
- Announcing to Solutions and TCS team
- Publish/Merge connectors 

Blocker:
- MySQL `AdvanceDataComparator`
